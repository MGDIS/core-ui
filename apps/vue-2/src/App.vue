<script>
import HelloWorld from './components/HelloWorld.vue';

export default {
  name: 'App',
  components: {
    HelloWorld,
  },
};
</script>

<template>
  <main role="main">
    <HelloWorld msg="Vue 2 app" />
  </main>
</template>
