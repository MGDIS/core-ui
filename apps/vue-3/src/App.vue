<script lang="tsx">
import { defineComponent } from 'vue';
import HelloWorld from './components/HelloWorld.vue';
import vueLogo from './assets/vue.svg';

export default defineComponent({
  render() {
    return (
      <>
        <div>
          <a href="https://vitejs.dev" target="_blank">
            <img src="/vite.svg" class="logo" alt="Vite logo" />
          </a>
          <a href="https://vuejs.org/" target="_blank">
            <img src={vueLogo} class="logo vue" alt="Vue logo" />
          </a>
        </div>
        <HelloWorld msg="Vite + Vue" />
      </>
    );
  },
});
</script>

<style scoped>
.logo {
  height: 6em;
  padding: 1.5em;
  will-change: filter;
  transition: filter 300ms;
}
.logo:hover {
  filter: drop-shadow(0 0 2em #646cffaa);
}
.logo.vue:hover {
  filter: drop-shadow(0 0 2em #42b883aa);
}
</style>
