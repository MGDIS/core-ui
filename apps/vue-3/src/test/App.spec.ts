import { mount } from '@vue/test-utils';
import { describe, expect, test } from 'vitest';
import App from '../App.vue';
import { axe } from 'jest-axe';

const getWrapper = () => mount(App);

describe('App', () => {
  test('Should render component', async () => {
    const wrapper = getWrapper();
    const html = wrapper.html();
    expect(await axe(html)).toHaveNoViolations();
    // Snapshot test
    expect(html).toMatchSnapshot();
  });
});
