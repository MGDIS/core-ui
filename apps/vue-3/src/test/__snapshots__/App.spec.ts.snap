// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > Should render component 1`] = `
<main role="main">
  <div>
    <a href="https://vitejs.dev" target="_blank">
      <img alt="Vite logo" class="logo" src="/vite.svg">
    </a>
    <a href="https://vuejs.org/" target="_blank">
      <img alt="Vue logo" class="logo vue" src="/src/assets/vue.svg">
    </a>
  </div>
  <h1>Vite + Vue</h1>
  <div class="card">
    <mg-button type="button" variant="secondary">count is 0</mg-button>
    <mg-modal close-button="true" hide="true" modal-title="Modal title">
      <div slot="content">
        <mg-form ariarole="presentation" identifier="test">
          <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
        </mg-form>
      </div>
    </mg-modal>
    <p>Edit<code>components/HelloWorld.vue</code> to test HMR</p>
  </div>
  <p>Check out<a href="https://vuejs.org/guide/quick-start.html#local" target="_blank">create-vue</a>, the official Vue + Vite starter</p>
  <p>Install<a href="https://github.com/vuejs/language-tools" target="_blank">Volar</a>in your IDE for a better DX</p>
  <p class="read-the-docs">Click on the Vite and Vue logos to learn more</p>
</main>
`;
